src/main/java/org/wso2/carbon/esb/connector/MQConfiguration.java

```diff
@@ -0,0 +1,398 @@
+/*
+*  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package org.wso2.carbon.esb.connector;
+
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static com.ibm.mq.constants.CMQC.*;
+
+/**
+ * IBM MQ configuration parameters
+ */
+public class MQConfiguration {
+
+    private final int port;
+    private final String host;
+    private String qManger;
+    private String topicName;
+    private String topicString;
+    private String queue;
+    private String channel;
+    private String userName;
+    private String password;
+    private int transportType;
+    private long timeout;
+    private int maxconnections;
+    private int maxunusedconnections;
+    private String ciphersuit;
+    private boolean flipRequired;
+    private boolean sslEnable;
+    private String trustStore;
+    private String trustPassword;
+    private String keyStore;
+    private String keyPassword;
+    private String messageID;
+    private String correlationID;
+    private String groupID;
+    private boolean persistent;
+    private int messageType = MQMT_DATAGRAM;
+    private String replyQueue;
+    private int priority;
+    private int replyTimeout;
+    private boolean islistenerEnabled;
+    private List<String> reconnectList = new ArrayList();
+    private int reconnectTimeout;
+    private List<String> channelList = new ArrayList<>();
+
+    MQConfiguration(MessageContext msg) {
+
+        if (msg.getProperty(MQConstants.PORT) != null) {
```

Comment:
EG:
 +        this.port = 1414;
 +        if (msg.getProperty(MQConstants.PORT) != null) {
 +            this.port = Integer.valueOf((String) msg.getProperty(MQConstants.PORT));
 +        }